File: explorer/templates/explorer_template.cpp
#include <iostream>

using namespace std;

int scan(int);
string submit(int);

/**
 * Perform scan queries and a submit query to find the length of the shortest
 * path from the vertex labeled 1 to the vertex labeled 1000 in the graph.
 */
void solve() {
    // YOUR CODE HERE
}

/**
 * Scan at the vertex labeled v. Returns the label of a random vertex that v is
 * connected to.
 */
int scan(int v) {
    cout << "SCAN " << v << endl;
    string response;
    cin >> response;
    if (response == "WRONG_ANSWER") {
        exit(0);
    }
    return stoi(response);
}

/**
 * Submit your guess for the length of the shortest path. Returns CORRECT if
 * your guess is correct and WRONG_ANSWER if your guess is wrong.
 */
string submit(int d) {
    cout << "SUBMIT " << d << endl;
    string response;
    cin >> response;
    if (response == "WRONG_ANSWER") {
        exit(0);
    }
    return response;
}

int main() {
    int T;
    cin >> T;
    for (int i = 0; i < T; ++i) {
        solve();
    }
    return 0;
}

File: explorer/templates/explorer_template.py
def solve() -> None:
    """
    Perform scan queries and a submit query to find the length of the shortest
    path from the vertex labeled 1 to the vertex labeled 1000 in the graph.
    """
    # YOUR CODE HERE


def scan(v: int) -> int:
    """
    Scan at the vertex labeled v. Returns the label of a random vertex that v is
    connected to.
    """
    print(f'SCAN {v}', flush=True)
    response = input()
    if response == 'WRONG_ANSWER':
        exit()
    return int(response)


def submit(d: int) -> str:
    """
    Submit your guess for the length of the shortest path. Returns CORRECT if
    your guess is correct and exits if your guess is wrong.
    """
    print(f'SUBMIT {d}', flush=True)
    response = input()
    if response == 'WRONG_ANSWER':
        exit()
    return response


def main():
    T = int(input())
    for _ in range(T):
        solve()
    verdict = input()


if __name__ == '__main__':
    main()
